File: Chalkable.BusinessLogic/Model/AcademicBenchmark/Authority.cs
using System;

namespace Chalkable.BusinessLogic.Model.AcademicBenchmark
{
    public class Authority
    {
        public Guid Id { get; set; }
        public string Code { get; set; }
        public string Description { get; set; }

        public static Authority Create(AcademicBenchmarkConnector.Models.Authority a)
        {
            return new Authority
            {
                Id = a.Id,
                Code = a.Code,
                Description = a.Description
            };
        }
    }
}

File: Chalkable.Web/app/chlk/models/academicBenchmark/Authority.js
REQUIRE('chlk.models.id.ABAuthorityId');

NAMESPACE('chlk.models.academicBenchmark', function(){

    /**@class chlk.models.academicBenchmark.Authority*/

    CLASS('Authority', [
        chlk.models.id.ABAuthorityId, 'id',
        String, 'code',
        String, 'description',

        function getName(){
            return this.getCode();
        }
    ]);
});
